front/src/common/Map/WarpedMap/getWarping.ts

```diff
@@ -0,0 +1,58 @@
+import bbox from '@turf/bbox';
+import simplify from '@turf/simplify';
+import { lineString } from '@turf/helpers';
+import { BBox2d } from '@turf/helpers/dist/js/lib/geojson';
+import { Feature, FeatureCollection, Geometry, LineString, Position } from 'geojson';
+
+import { extendLine, getSamples } from 'common/Map/WarpedMap/core/helpers';
+import { getQuadTree } from 'common/Map/WarpedMap/core/quadtree';
+import {
+  featureToPointsGrid,
+  getGridIndex,
+  getGrids,
+  pointsGridToZone,
+  straightenGrid,
+} from 'common/Map/WarpedMap/core/grids';
+import { clipAndProjectGeoJSON, projectBetweenGrids } from 'common/Map/WarpedMap/core/projection';
+
+export type WarpingFunction = ReturnType<typeof getWarping>['transform'];
+
+export default function getWarping(warpPath: Feature<LineString>) {
+  // Simplify the input path to get something "straighter", so that we can see
+  // in the final warped map the small curves of the initial path:
+  // TODO: Detect loops and remove them from the simplifiedPath
+  const simplifiedPath = simplify(warpPath, { tolerance: 0.01 });
+
+  // Cut the simplified path as N equal length segments
+  const sample = getSamples(simplifiedPath, 15);
+  const samplePath = lineString(sample.points.map((point) => point.geometry.coordinates));
+
+  // Extend the sample, so that we can warp things right before and right
+  // after the initial path:
+  const extendedSamples = extendLine(samplePath, sample.step);
+  const steps = extendedSamples.geometry.coordinates.length - 1;
```

Comment:
stepCount/stepsCount is a better name I think
```suggestion
  const stepsCount = extendedSamples.geometry.coordinates.length - 1;
```